### PR TITLE
refactor(@angular/cli): simplify default arguments in angular-cli.json

### LINKED ISSUE #4137: Tracking: Styleguide Compliance
Modifications to angular-cli.json (including schema and TS definition file)

- [x] Reorganize by generation/blueprint (currently by configuration option)

Ensure support of the following options:

- [x] spec generation
- [x] flat folder
- [x] inline template (component only)
- [x] inline style (component only)
- [x] view encapsulation (component only)
- [x] change detection (component only)

### PR DESCRIPTION
**Pull request by Brocco**
Fixes #4137

### CODE REVIEW

File: packages/@ngtools/json-schema/src/schema-class-factory.ts [hunk 1]
```diff
@@ -117,6 +117,10 @@ class SchemaClassBase<T> implements SchemaClass<T> {
   /** Get a value from a JSON path. */
   $$get(path: string): any {
     const node = _getSchemaNodeForPath(this.$$schema(), path);
+    if (path === 'defaults.component.viewEncapsulation') {
```
Comment (hansl) [inline packages/@ngtools/json-schema/src/schema-class-factory.ts]: tsk tsk

File: packages/@ngtools/json-schema/src/schema-tree.ts [hunk 2]
```diff
@@ -406,6 +406,7 @@ export abstract class LeafSchemaTreeNode<T> extends SchemaTreeNode<T> {
 
   get() {
     if (!this.defined && this._forward) {
+      console.log('fwd');
```
Comment (hansl) [inline packages/@ngtools/json-schema/src/schema-tree.ts]: tsk tsk

File: packages/@angular/cli/blueprints/class/index.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export default Blueprint.extend({
 
     options.spec = options.spec !== undefined ?
       options.spec :
-      this.project.ngConfigObj.get('defaults.spec.class');
+      this.project.ngConfigObj.get('defaults.class.spec');
```
Comment (hansl) [inline packages/@angular/cli/blueprints/class/index.ts]: Please use an alias for these, and a warning if the alias is used.

See [here](https://github.com/angular/angular-cli/blob/0dc2200004a5621a1576624b527e529a1bb1ecea/packages/%40angular/cli/models/config.ts#L50) for an example.

**Comment by Brocco** [comment]
@hansl this PR has been updated addressing the requested changes.

File: packages/@angular/cli/models/config.ts [hunk 4]
```diff
@@ -52,6 +52,17 @@ export class CliConfig extends CliConfigBase<ConfigInterface> {
...
+    cliConfig.alias('defaults.component.inlineTemplate', 'defaults.inline.template');
+    cliConfig.alias('defaults.component.spec', 'defaults.spec.component');
+    cliConfig.alias('defaults.class.spec', 'defaults.spec.class');
+    cliConfig.alias('defaults.component.directive', 'defaults.spec.directive');
+    cliConfig.alias('defaults.component.module', 'defaults.spec.module');
+    cliConfig.alias('defaults.component.pipe', 'defaults.spec.pipe');
+    cliConfig.alias('defaults.component.service', 'defaults.spec.service');
+
     // If any of them returned true, output a deprecation warning.
     if (aliases.some(x => !!x)) {
```
Comment (filipesilva) [inline packages/@angular/cli/models/config.ts]: This warning should also be updated. Maybe it should be a generic warning? Not sure. I have a PR that will also need to do this.

Comment (Brocco) [inline packages/@angular/cli/models/config.ts]: @hansl and I chatted and we decided to leave the remapping of these as silent, as the correct values will still be handled correctly in the config file.

Comment (filipesilva) [inline packages/@angular/cli/models/config.ts]: I'll do the same to mine then.

**Pull request merged**